### PR TITLE
Describe and match GUI

### PR DESCRIPTION
**Pull request by rperrot**
Hi, 

this PR adds a (Qt) GUI for computing description and matching on a pair of images using the description/matching algorithms provided by openMVG. This can be a demonstrator of some parts of openMVG. 

It's known to work with Qt 5.8.

Hope it helps !

### CODE REVIEW

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  void onExportImage( void );
+
+private:
+  /**
+  * @brief Build all interface widgets 
+  */
+  void BuildInterface();
+
+  /**
+  * @brief Build all menus items
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: menu items

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  */
+  void BuildMenus();
+
+  /**
+  * @brief Make connections between interface elements
+  */
+  void MakeConnections();
+
+  /** 
+  * @brief Get list of all feature type (and mode)
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: 'matching types"?
PopulateFeatureTypes?

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  */
+  void PopulateFeatureType();
+
+  /**
+  * @brief Get list of all matching type
+  */
+  void PopulateMatchingType();
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: idem

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+  /**
+  * @brief remove previous computation on the view (only view data, not internal ones)
+  */
+  void ClearFeaturesAndMatchItems();
+
+  /**
+  * @brief Redraw match lines at correct position (ie:after image move)
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: "the correct position" + "i.e."

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  */
+  void MoveMatcheLines();
+
+  /**
+  * @brief Get feature preset from interface
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: "from the interface"

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  * @return descriptor preset given the interface choice 
+  */
+  openMVG::features::EDESCRIBER_PRESET GetFeaturePreset( void );
+
+  /**
+  * @brief Get describer instance based on interface choice
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: "the choices made in the interface"

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  /**
+  * @brief Get describer instance based on interface choice 
+  * @param preset Current preset 
+  * @return An image describer 
+  */
+  std::unique_ptr<openMVG::features::Image_describer> GetFeatureDescriber( const openMVG::features::EDESCRIBER_PRESET preset );
+
+  /**
+  * @brief Get Matcher type 
+  * @return current matcher type based on interface choices
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.hpp]: idem

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.cpp [hunk 8]
```diff
@@ -0,0 +1,553 @@
...
+
+/**
+  * @brief Action to be done when user wants to compute features and matching bewteen image pairs 
+  */
+void MainWindow::onComputeMatching( void )
+{
+
+  if ( !( m_image1Path.size() > 0 && m_image2Path.size() > 0 ) )
+  {
+    // TODO : add a message box to tell that we want two image before processing
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.cpp]: two images

File: src/openMVG_Samples/describe_and_match_GUI/MainWindow.cpp [hunk 9]
```diff
@@ -0,0 +1,553 @@
...
+
+  progress.setValue( 2 );
+
+  // 2.1 Get matching params
+  const EMatcherType matcherType = GetMatcherType();
+  const float fDistRatio         = static_cast<float>( m_spinDistThreshold->value() );
+
+  m_matches.clear();
+
+  // 2.1 Compute matching
```
Comment (rjanvier) [inline src/openMVG_Samples/describe_and_match_GUI/MainWindow.cpp]: "Match features/Match images" should be better

**Comment by pmoulon** [comment]
Thanks, will merge it soon your PR to develop ;-)

**Comment by pmoulon** [comment]
Thanks @rperrot :+1: 
Merged thanks to https://github.com/openMVG/openMVG/commit/8f784a020eab7e29aaab3e193ebc542cf2764559 and https://github.com/openMVG/openMVG/commit/64a163707ced858b89f8f67c616bfb866b3ff902

Note: I observed a problem on Linux, the progress bar does not appear.

**Pull request closed without merging**